src/sentry/search/snuba/backend.py

```diff
@@ -248,7 +248,8 @@ def _query(self, project, retention_window_start, group_queryset, tags, environm
         # and the result groups are then post-filtered via queries to the Sentry DB
         optimizer_enabled = options.get('snuba.search.pre-snuba-candidates-optimizer')
         if optimizer_enabled:
-            key = 'snuba.search:project.group.count:%s' % project.id
+            key = 'snuba.search:project.group.count:%s' % ','.join(
+                [six.text_type(p.id) for p in projects])
```

Comment:
We might want to hash the concatenated project ids for use in this key. Since we're using memcached as our cache backend, the max key length is 250 characters, and we could easily overrun that with a lot of projects

Reply:
Alternatively you could make the snuba query `GROUP BY project_id` to get individual counts for all the projects and set those keys in memcached, then up here you can try and multi-get the values for each project in memcached and sum the results. That way the cache is more reusable if you change the selected projects slightly

Reply:
Using something like an `md5(','.join(projectids))` would help with preventing key overflows.